contracts/rewards/RewardsDistributor.sol

```diff
@@ -169,303 +237,275 @@ abstract contract RewardsDistributor is IRewardsDistributor {
       }
 
       // Due emissions is still zero, updates only latestUpdateTimestamp
-      _updateAssetStateInternal(
-        rewardsInput[i].asset,
-        rewardsInput[i].reward,
+      (uint256 newIndex, ) = _updateRewardData(
         rewardConfig,
         rewardsInput[i].totalSupply,
-        _assets[rewardsInput[i].asset].decimals
+        10**decimals
       );
 
       // Configure emission and distribution end of the reward per asset
+      uint88 oldEmissionsPerSecond = rewardConfig.emissionPerSecond;
+      uint32 oldDistributionEnd = rewardConfig.distributionEnd;
       rewardConfig.emissionPerSecond = rewardsInput[i].emissionPerSecond;
       rewardConfig.distributionEnd = rewardsInput[i].distributionEnd;
 
       emit AssetConfigUpdated(
         rewardsInput[i].asset,
         rewardsInput[i].reward,
+        oldEmissionsPerSecond,
         rewardsInput[i].emissionPerSecond,
-        rewardsInput[i].distributionEnd
+        oldDistributionEnd,
+        rewardsInput[i].distributionEnd,
+        newIndex
       );
     }
   }
 
   /**
-   * @dev Updates the state of one distribution, mainly rewards index and timestamp
-   * @param asset The address of the asset being updated
-   * @param reward The address of the reward being updated
-   * @param rewardConfig Storage pointer to the distribution's reward config
+   * @dev Updates the state of the distribution for the specified reward
+   * @param rewardData Storage pointer to the distribution reward config
    * @param totalSupply Current total of underlying assets for this distribution
-   * @param decimals The decimals of the underlying asset
+   * @param assetUnit One unit of asset (10**decimals)
    * @return The new distribution index
+   * @return True if the index was updated, false otherwise
    **/
-  function _updateAssetStateInternal(
-    address asset,
-    address reward,
-    RewardData storage rewardConfig,
+  function _updateRewardData(
+    RewardsDataTypes.RewardData storage rewardData,
     uint256 totalSupply,
-    uint8 decimals
-  ) internal returns (uint256) {
-    uint256 oldIndex = rewardConfig.index;
-
-    if (block.timestamp == rewardConfig.lastUpdateTimestamp) {
-      return oldIndex;
-    }
-
-    uint256 newIndex = _getAssetIndex(
-      oldIndex,
-      rewardConfig.emissionPerSecond,
-      rewardConfig.lastUpdateTimestamp,
-      rewardConfig.distributionEnd,
-      totalSupply,
-      decimals
-    );
-
-    if (newIndex != oldIndex) {
+    uint256 assetUnit
+  ) internal returns (uint256, bool) {
+    (uint256 oldIndex, uint256 newIndex) = _getAssetIndex(rewardData, totalSupply, assetUnit);
+    bool dataUpdated;
+    if ((dataUpdated = newIndex != oldIndex)) {
```

Comment:
`dataUpdated` should probably be `indexUpdated` as the timestamp is still updated when the value is false.